examples/thermostat/silabs/src/SensorManager.cpp

```diff
@@ -81,19 +82,20 @@ void SensorManager::SensorTimerEventHandler(void * arg)
     int16_t temperature            = 0;
     static int16_t lastTemperature = 0;
 
-#ifdef USE_TEMP_SENSOR
-    int32_t tempSum   = 0;
-    uint32_t humidity = 0;
+#if defined(SL_MATTER_USE_SI70XX_SENSOR) && SL_MATTER_USE_SI70XX_SENSOR
+    int16_t tempSum   = 0;
```

Comment:
we are taking 100 samples of decidegree Celsius, we could overflow the int16_t. You should keep the int32_t